### ISSUE TITLE
Parsing error on &!

### ISSUE THREAD

**Issue by AndrewKvalheim** [abathur/resholve#85]
zsh has a [`&!`](https://zsh.sourceforge.io/Doc/Release/Jobs-_0026-Signals.html) syntax for starting a disowned job, e.g.:

```shell
sleep 3s &!
```

Using this in a script causes resholve to fail with a parsing error:

```console
$ resholve --interpreter "$SHELL" --path '' <<< 'sleep 3s &!'
  sleep 3s &!
             ^
[ stdinNone ]:2: error: Invalid word while parsing command
```

**Comment by abathur** [comment]
This is expected (for now, at least). It's documented in the manpage (here's an excerpt from the plaintext copy):

https://github.com/abathur/resholve/blob/80e93633c824175e8afb25b0fa02aee35203e6fc/docs/resholve.1.txt#L202-L213

In the near-ish term, we can probably:
- [ ] Use the documentation-generation capabilities earlier this year to finally start generating the main README.md, which should make it easier to keep it (and other doc outputs) in sync on some questions like this.
- [ ] Look into decorating parse errors that from the Oil/OSH parser with a little more context on this point.

In the long run I would _like_ for resholve to be a little more parser-agnostic, but my gut says that'll be pretty difficult. I made some strides over the last year to build a more domain-specific abstraction layer between resholve and the OSH parser, but it is still tightly coupled to some of Oil's abstractions and APIs.

**Comment by AndrewKvalheim** [comment]
Thanks, makes sense. Indeed, I’d only been looking at the main and nixpkgs readmes. For now I’ll just use a wrapper around resholve to temporarily replace `&!` with `@AMPERSAND_EXCLAMATION@`.

**Comment by abathur** [comment]
Good point--if pluggable parsers don't seem to be tractable, I guess pluggable in/out filters might be...
